.github/workflows/release-packages.yaml

```diff
@@ -130,6 +130,28 @@ jobs:
           $msi_name = Get-ChildItem -Filter *.msi -Name
           Write-Output "::set-output name=msi_installer::build/$msi_name"
           Write-Output "::set-output name=msi_name::$msi_name"
+      - name: Decode signing certificate
+        id: decode_certificate
+        run: |
+          $pfx_filename=[string](Get-Location)+"\CodeSignCertificate.pfx"
+          $pfx_bytes=[System.Convert]::FromBase64String("${{ secrets.CODESIGNCERTPFX }}")
+          [IO.File]::WriteAllBytes($pfx_filename, $pfx_bytes)
+      - name: Sign the installer
+        id: code_sign
+        run: |
+          $pfx_filename=[string](Get-Location)+"\CodeSignCertificate.pfx"
+          $signtool = $(Get-ChildItem -Path ${env:ProgramFiles(x86)} -Recurse -Filter 'signtool.exe' | Where-Object FullName -like '*10.0.19041.0\x64\signtool.exe').FullName
```

Comment:
I don’t like that this simultaneously hardcodes a path but then still does a search for it. Are you sure there’s no easier way to get at this path? I’m thinking in particular of `vswhere` here, which we already use to find `msbuild` in the `setup-msbuild` action above.

Reply:
To be absolutely honest I have not checked singtool by running only signtool.exe, there are multiple versions of Windows Software Development Kit installed on the Github Actions Windows agent with signtool.exe for each of them and since I was testing it offline on latest SDK (version 10.0.19041.0) I decided to go for this approach with search for specific version. Additionally there are multiple platforms supported by signtool (x64, x86, arm, etc.). Maybe the default singtool.exe will be the correct one - I will check if signtool is in the path and in which version but I am afraid that the default one may not always be the correct one.

I have tried to find a way of detecting signtool with vswhere but I could not get anywhere with it - signtool.exe comes from Windows SDK not directly from VS package so maybe it is not possible to locate it with vswhere.

Reply:
Unfortunately there is no `signtool.exe` in the path, I can either leave it as is (with the search), or hard-code the "Program Files" path - I personally would prefer to give it a little bit of flexibility and let the search sort the path out. Also I could not find anything about getting `signtool` path through `vswere`.

Reply:
While I'd strongly favour a keeping-it-simple approach, could we at bare minimum only do this once? I think https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files should help here.

Reply:
@tautschnig Thanks for your suggestion! it was not super straight-forward but I have managed to get the workflow simplified a little bit with the Github actions environmental files. 
There is a new job in the workflow: "Setup code sign environment" which is appending the path for `signtool.exe` to `env:GITHUB_PATH` file, and storing PFX certificate file name as "`pfxcert`" variable in `env:GITHUB_ENV` file. Following jobs are now referring to `$env:pfxcert` instead of concatenating the PFX file name every time it is needed and the `signtool.exe` is accessible without the previously required search.